cli/script/command-executor.ts

```diff
@@ -1091,88 +1092,132 @@ function patch(command: cli.IPatchCommand): Promise<void> {
 }
 
 export var release = (command: cli.IReleaseCommand): Promise<void> => {
-    if (isBinaryOrZip(command.package)) {
+    if (isBinaryOrZip(command.path)) {
         throw new Error("It is unnecessary to package releases in a .zip or binary file. Please specify the direct path to the update content's directory (e.g. /platforms/ios/www) or file (e.g. main.jsbundle).");
     }
 
     throwForInvalidSemverRange(command.appStoreVersion);
-    var filePath: string = command.package;
-    var getPackageFilePromise: Promise<IPackageFile>;
-    var isSingleFilePackage: boolean = true;
 
-    if (fs.lstatSync(filePath).isDirectory()) {
-        isSingleFilePackage = false;
-        getPackageFilePromise = Promise<IPackageFile>((resolve: (file: IPackageFile) => void, reject: (reason: Error) => void): void => {
-            var directoryPath: string = filePath;
+    // Copy the command so that the original is not modified
+    var currentCommand: cli.IReleaseCommand = {
+        appName: command.appName,
+        appStoreVersion: command.appStoreVersion,
+        deploymentName: command.deploymentName,
+        description: command.description,
+        disabled: command.disabled,
+        mandatory: command.mandatory,
+        path: command.path,
+        rollout: command.rollout,
+        signingKeyPath: command.signingKeyPath,
+        type: command.type
+    };
+
+    var hooks: cli.ReleaseHook[] = [ signingReleaseHook.default, coreReleaseHook ];
+
+    var releaseHooksPromise = hooks.reduce((accumulatedPromise: Q.Promise<cli.IReleaseCommand>, hook: cli.ReleaseHook) => {
+        return accumulatedPromise
+            .then((modifiedCommand: cli.IReleaseCommand) => {
+                currentCommand = modifiedCommand || currentCommand;
```

Comment:
Could you give some context into what we are trying to do here? Why will the command need to be modified?